### PR TITLE
Fix parameter parsing when its value contains equal symbol

### PR DESCRIPTION
**Pull request by ans0600**
Make sure external parameter is set correctly when its value contains "="

### CODE REVIEW

**Comment by gergnz** [comment]
Thanks @ans0600. Can you give me an example of when this might be an issue?

**Comment by ans0600** [comment]
HI @gergnz ,

We need to pass an url with query string as parameter and it will break set set_param function. Overall, I think this could increase the robustness of this feature.

Another case I can think of is some API key will be base64 encoded so it is very likely to contain "=".

File: lib/cfndsl.rb [hunk 1]
```diff
@@ -74,7 +74,8 @@ def self.eval_file_with_extras(filename, extras = [], logstream = nil)
           b.eval(File.read(file), file)
         end
       when :raw
-        params.set_param(*file.split('='))
+        file_parts = file.split("=")
+        params.set_param(file_parts[0],file_parts[1..-1].join(""))
```
Comment (gergnz) [inline lib/cfndsl.rb]: Should the join have "=" in it?

**Comment by gergnz** [comment]
Hi @ans0600,

Thanks for the PR. I'm not sure the test coverage is complete enough, and I'm not 100% sure you've achieved what you set out to achieve.

Let me know if I'm on the wrong path, but here's my example:
```ruby
# frozen_string_literal: true

require 'cfndsl'

CloudFormation do
  Parameter(:Foobar) do
    Type 'String'
    Default external_parameters[:foobar]
  end
  Parameter(:Dingbat) do
    Type 'String'
    Default external_parameters[:dingbat]
  end
end
```

And I call that with:
```bash
cfndsl -b -p -D foobar="adsafsa=adsf" -D dingbat="adfsdasf===009=99" x.rb
```

Which gives me:
```javascript
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Parameters": {
    "Foobar": {
      "Type": "String",
      "Default": "adsafsaadsf"
    },
    "Dingbat": {
      "Type": "String",
      "Default": "adfsdasf00999"
    }
  }
}
```

All my _=_ signs are missing.

If this should have my _=_ signs, then I think you need an _=_ in the join, but it also highlights that your rspec test is incomplete.

I'm not 100% sure this is the issue though. Sorry if I'm misunderstanding the problem.

**Comment by ans0600** [comment]
@gergnz oops, it should join with "=". Thanks very much for your detailed feedback and testing.

I have also updated the test to evaluate the output template regarding the external value. Please have a look.

**Comment by gergnz** [comment]
Cool. Thanks @ans0600. That looks more complete.

**Pull request merged**